rioxarray/xarray_plugin.py

```diff
@@ -0,0 +1,20 @@
+import os.path
+
+import xarray as xr
+
+from . import _io
+
+
+class GdalBackend(xr.backends.common.BackendEntrypoint):
+    def open_dataset(self, filename_or_obj, drop_variables=None):
+        ds = _io.open_rasterio(filename_or_obj).to_dataset("band")
+        ds = ds.rename({idx: f"band{idx}" for idx in ds.data_vars})
+        ds = ds.reset_coords("spatial_ref")
+        return ds
+
+    def guess_can_open(self, filename_or_obj):
+        try:
+            _, ext = os.path.splitext(filename_or_obj)
+        except TypeError:
+            return False
+        return ext in {".tif", ".geotif", ".tiff", ".geotiff"}
```

Comment:
Just had this thought. With rasterio 1.2+ you can do:
```python
>>> from rasterio.drivers import raster_driver_extensions
>>> list(raster_driver_extensions())
['vrt', 'tif', 'tiff', 'ntf', 'img', 'asc', 'dt0', 'dt1', 'dt2', 'png', 'jpg', 'jpeg', 'gif', 'xpm', 'bmp', 'pix', 'map', 'mpr', 'mpl', 'rgb', 'hgt', 'ter', 'nc', 'lbl', 'cub', 'xml', 'ers', 'jp2', 'j2k', 'grb', 'grb2', 'grib2', 'rsw', 'rst', 'grd', 'rda', 'kml', 'kmz', 'webp', 'pdf', 'sqlite', 'mbtiles', 'cal', 'ct1', 'mrf', 'pgm', 'ppm', 'pnm', 'hdr', 'bt', 'lcp', 'gtx', 'gsb', 'gvb', 'kro', 'byn', 'err', 'dem', 'kea', 'bag', 'gen', 'blx', 'sdat', 'sg-grd-z', 'xyz', 'hf2', 'dat', 'sigdem', 'gpkg', 'bil']
```

Reply:
That's a long list! 😆 

We need to be careful as we don't want to override netCDF, GRIB or other data formats best accessed with a dedicated backend, so I would still keep a curated list of extensions that `rioxarray` offers to open without passing the `engine`.

Reply:
I am not familiar with how `xarray` selects the driver they want. Is there a way for `xarray` to set the preferred order to try to open the dataset?

Reply:
There is no such feature at the moment, the order of testing is not specified and the first plugin that matches is selected.

Anyway the `guess_can_open` is only used when `engine` is not passed explicitly, so it is a convenience feature aimed at non-expert users.

I think it is safe to add to `CAN_OPEN_EXT` the extensions of GIS file formats, but not more scientific formats that have, ora are likely to have a dedicated plugin.